rasa/shared/core/trackers.py

```diff
@@ -425,7 +425,10 @@ def applied_events(self) -> List[Event]:
                     event.action_name, applied_events
                 )
             else:
-                applied_events.append(event)
+                if (len(applied_events) == 0) and isinstance(event, SlotSet):
```

Comment:
I think this should be addressed at the level where the SlotSet events are being added to the tracker, not when they're retrieved

Reply:
Can you explain that suggestion, as I would not recommend changing how the slots are put on the tracker ? and technically I think it is the way the events are pulled from the tracker and used in the TED policy in this unique scenario , can you confirm that ?